merlin/common/tasks.py

```diff
@@ -435,24 +442,38 @@ def expand_tasks_with_samples(
         for step in steps
     ]
 
-    workspaces = [step.get_workspace() for step in steps]
-    LOG.debug(f"workspaces : {workspaces}")
+    #workspaces = [step.get_workspace() for step in steps]
+    #LOG.debug(f"workspaces : {workspaces}")
 
     needs_expansion = is_chain_expandable(steps, labels)
 
+    LOG.debug(f"needs_expansion {needs_expansion}")
+
     if needs_expansion:
-        prepare_chain_workspace(sample_index, steps)
+        #prepare_chain_workspace(sample_index, steps)
         sample_index.name = ""
-        LOG.debug(f"queuing merlin expansion task")
-        sig = add_merlin_expanded_chain_to_chord.s(
-            task_type, steps, samples, labels, sample_index, adapter_config, 0
-        )
-        sig.set(queue=steps[0].get_task_queue())
-        if self.request.is_eager:
-            sig.delay()
-        else:
-            self.add_to_chord(sig, lazy=False)
-        LOG.debug(f"merlin expansion task queued")
+        LOG.debug(f"queuing merlin expansion tasks")
+        for next_index_path, next_index in sample_index.traverse(conditional = lambda c: c.is_great_grandparent_of_leaf):
```

Comment:
This loop is not being executed in feature_demo.

Reply:
ok i found the reproducer

this fails for n_samples <= number of directories per level

Reply:
the problem looks to be that if there's a flat hierarchy, this doesn't trigger queueing anything off